packages/loader/container-loader/src/container.ts

```diff
@@ -169,14 +170,15 @@ export enum ConnectionState {
  */
 export async function waitContainerToCatchUp(container: IContainer) {
     // Make sure we stop waiting if container is closed.
+    const containerClosedMessage = "Container is closed";
     if (container.closed) {
-        throw new Error("Container is closed");
+        throw new Error(containerClosedMessage);
     }
 
     return new Promise<boolean>((resolve, reject) => {
         const deltaManager = container.deltaManager;
 
-        container.on("closed", reject);
+        container.on("closed", (err?: IErrorBase | undefined) => reject(err ?? new Error(containerClosedMessage)));
```

Comment:
How about:

```suggestion
        container.on("closed", (err?: IErrorBase | undefined) => reject(wrapError((m)=>new GenericError(`Container closed while waiting to catch up: ${m}`)));
```

Excuse the typos, I'm on my phone :)

This way the message is consistent at least in the prefix, regardless of whether the container closed due to an error or just by the user. If there's no error then I think m will be "undefined", which is a little weird so feel free to iterate on this. I've just found it hard to track what's happening when doing err ?? new Error(...) since you have no clue what err could be.

Also, we try to have errors coming out of our top-level APIs have errorType. This will have errorType "genericError".

Reply:
Yeah, this seems better. Thanks! I didn't know about UsageError either; will have to go through the types in that file for my own learning.